src/R/Components/Impl/ConnectionManager/Implementation/Connection.cs

```diff
@@ -20,5 +30,43 @@ internal class Connection : ConnectionInfo, IConnection {
         /// If true, the connection is to a remote machine
         /// </summary>
         public bool IsRemote { get; }
+
+        public override string UserProvidedPath {
+            get { return base.UserProvidedPath; }
+            set {
+                base.UserProvidedPath = value;
+                Path = ToCompletePath(value);
+            }
+        }
+
+        public static string ToCompletePath(string path) {
```

Comment:
The latter is better IMO. Good example is mobile Safari. Type `CNN` in IE or Chrome and they take you to search. Ty CNN in Safari and it takes you there. It doesn't add `http` or anything - does add .com since you may want to be able to tell site.com from site.org.

Although I'd rather leave user data alone. Not everyone appreciate forced education ;-)